File: java/src/main/java/io/jkratz/katas/gameoflife/GameOfLife.java
package io.jkratz.katas.gameoflife;

public class GameOfLife {

    public static void main(String[] args) {
        final Grid grid = new Grid();
        int[][] current = grid.getState();
        grid.nextState();
        int[][] nextState = grid.getState();

        System.out.println("Current Generation");
        printGrid(current);
        System.out.println("Next Generation");
        printGrid(nextState);
    }

    // TODO: This could be placed in a better location
    private static void printGrid(int[][] grid) {
        for (int[] columns : grid) {
            for (int value : columns) {
                if (value == 0) {
                    System.out.print(" . ");
                } else {
                    System.out.print(" * ");
                }
            }
            System.out.println();
        }
    }
}

File: kotlin/src/main/kotlin/io/jkratz/katas/gameoflife/GameOfLife.kt
package io.jkratz.katas.gameoflife

object GameOfLife {

    @JvmStatic
    fun main(args: Array<String>) {
        val grid = Grid()
        val current = grid.state
        grid.nextState()
        val nextState = grid.state

        println("Current Generation")
        printGrid(current!!)
        println("Next Generation")
        printGrid(nextState!!)
    }

    // TODO: This could be placed in a better location
    private fun printGrid(grid: Array<IntArray>) {
        for (columns in grid) {
            for (value in columns) {
                if (value == 0) {
                    print(" . ")
                } else {
                    print(" * ")
                }
            }
            println()
        }
    }
}